helix-rest/src/test/java/org/apache/helix/rest/client/TestCustomRestClient.java

```diff
@@ -145,6 +148,35 @@ public void testPostRequestFormat() throws IOException {
     Assert.assertEquals(json.get("data").asText(), "{}");
   }
 
+  @Test
+  public void testGetInstanceStoppableCheckSmallTimeout() throws Exception {
+    // Reset the INSTANCE in CustomRestClientFactory to create new one with new TimeOut
+    Field instance = CustomRestClientFactory.class.getDeclaredField("INSTANCE");
+    instance.setAccessible(true);
+    instance.set(null, null);
+
+    // Set 1 ms to cause timeout for http requests
+    System.setProperty(RestSystemPropertyKeys.HTTP_TIMEOUT_MS, "1");
+    // a popular echo server that echos all the inputs
+    // TODO: add a mock rest server
+    final String echoServer = "http://httpbin.org/post";
+    CustomRestClient _customRestClient = CustomRestClientFactory.get();
+    boolean timeoutExceptionHappened = false;
+    try {
+      _customRestClient.getInstanceStoppableCheck(echoServer, Collections.emptyMap());
+    } catch (ConnectTimeoutException e) {
+      // Since the timeout is so small, we are expecting to get ConnectTimeoutException
+      timeoutExceptionHappened = true;
+    }
+    // Reset the HTTP_REQUEST_TIMEOUT property back to the default value
+    System.setProperty(RestSystemPropertyKeys.HTTP_TIMEOUT_MS, "60000");
```

Comment:
Why is this reset to a hard coded timeout? What if the timeout is changed, you setting it here 60000 would break the original value. The correct way should be to backup the original property and set it back here.

Reply:
Should use try finally to recover the state. Otherwise if this test throws exception before recovering, later tests may be failing.

Reply:
Removed this test.